src/borg/archive.py

```diff
@@ -700,22 +713,39 @@ def process_symlink(self, path, st):
         self.add_item(item)
         return 's'  # symlink
 
+    def chunk_file(self, item, cache, stats, fd, fh=-1, **chunk_kw):
+        checkpoint_number = 1
+        item.chunks = []
+        for data in backup_io_iter(self.chunker.chunkify(fd, fh)):
+            item.chunks.append(cache.add_chunk(self.key.id_hash(data), Chunk(data, **chunk_kw), stats))
+            if self.show_progress:
+                self.stats.show_progress(item=item, dt=0.2)
+            if self.checkpoint_interval and time.time() - self.last_checkpoint > self.checkpoint_interval:
+                checkpoint_item = Item(internal_dict=item.as_dict())
+                checkpoint_item.path += '.checkpoint_%d' % checkpoint_number
+                checkpoint_item.checkpoint = checkpoint_number
+                checkpoint_number += 1
+                self.add_item(checkpoint_item, show_progress=False)
```

Comment:
@enkore did you consider that? ^^^

That means that there will be following items in the metadata stream:
```
filename/ITEM    chunks
---------------------------------------------------------
foo.checkpoint_1 abc
foo.checkpoint_2 abcdef
foo              abcdefghi
---------------------------------------------------------
refcount         333222111
```
While the checkpoint ARCHIVES get removed, the item metadata stream is always being appended, so the checkpoint ITEMS stay (the final, finishes archive has all the checkpoint ITEMS).